File: java/BillsApi/src/main/java/com/apilangarena/javabillsapi/application/bills/get/GetBillsService.java
package com.apilangarena.javabillsapi.application.bills.get;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GetBillsService {
    private final BillReadRepository billReadRepository;

    public GetBillsService(BillReadRepository billReadRepository) {
        this.billReadRepository = billReadRepository;
    }

    public List<BillSummaryDto> execute() {
        return billReadRepository.listBills();
    }
}

File: kotlin/BillsApi/src/main/kotlin/com/apilangarena/billsapi/application/bills/get/GetBillsService.kt
package com.apilangarena.billsapi.application.bills.get

import org.springframework.stereotype.Service

@Service
class GetBillsService(
    private val billReadRepository: BillReadRepository,
) {
    fun execute(): List<BillSummaryDto> = billReadRepository.listBills()
}
